minoc/src/main/java/org/opencadc/minoc/PutAction.java

```diff
@@ -164,21 +165,27 @@ public void doAction() throws Exception {
         NewArtifact newArtifact = new NewArtifact(artifactURI);
         newArtifact.contentChecksum = contentMD5;
         newArtifact.contentLength = contentLength;
-        
+
+        final Profiler putProfiler = new Profiler(PutAction.class);
+
         InputStream in = (InputStream) syncInput.getContent(INLINE_CONTENT_TAG);
-        
+
+        putProfiler.checkpoint("getContent(inputstream)");
+
         StorageMetadata artifactMetadata = null;
         
         log.debug("writing new artifact to storage...");
         try {
             artifactMetadata = getStorageAdapter().put(newArtifact, in);
+            putProfiler.checkpoint("getStorageAdapter().put() success.");
```

Comment:
Would be good to separate the getStorageAdapter() call from put() and add an extra profiler call between them, which could be non-zero in some implementations.

storageAdapter.init
storageAdapter.put.ok